packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -489,6 +491,9 @@ abstract class ImageStreamCompleter with Diagnosticable {
   /// if all [keepAlive] handles get disposed.
   bool _hadAtLeastOneListener = false;
 
+  /// Whether the listeners should be called synchronously
+  bool synchronousCall = true;
```

Comment:
this should be private

Reply:
Unfortunately, the ImageStreamCompleter doesn't know whether the synchronousCall parameter should be set to true or not. That's why the ImageStream needs to make the completer aware of it; either by passing it in as a parameter to the addListener function or through a property.

Reply:
ImageStream should be able to access the private variable in ImageStreamCompleter because they are in the same file.

Reply:
also maybe renaming this to _addingInitialListener to be more self-explanatory

Reply:
I was not aware this was a thing, thanks!